weave-js/src/components/PagePanelComponents/Home/Browse3/charts/persistence.ts

```diff
@@ -0,0 +1,67 @@
+/*
+  persistence.ts
+
+  This file contains functions for saving and loading charts configuration from localStorage.
+*/
+import {v4 as uuidv4} from 'uuid';
+
+import {ChartConfig} from './types';
+
+export const LOCAL_STORAGE_KEY_PREFIX = 'callsChartsConfig';
+
+function getContextualStorageKey(entity: string, project: string) {
+  return `${LOCAL_STORAGE_KEY_PREFIX}_${entity}_${project}`;
+}
+
+function getDefaultChartsConfig() {
+  return {
+    charts: [
+      {
+        id: uuidv4(),
+        xAxis: 'started_at',
+        yAxis: 'cost',
+        plotType: 'bar',
+        binCount: 20,
+        aggregation: 'sum',
+      },
+      {
+        id: uuidv4(),
+        xAxis: 'started_at',
+        yAxis: 'latency',
+        plotType: 'line',
+        binCount: 20,
+        aggregation: 'p95',
+      },
+      {
+        id: uuidv4(),
+        xAxis: 'prompt_tokens',
+        yAxis: 'completion_tokens',
+        plotType: 'scatter',
+      },
+    ],
+    openDrawerChartId: null,
+  };
+}
+
+export function loadChartsConfig(
```

Comment:
even thought these are not async right now - once they are backed by backend data storage, they will need to be. Making them async now could help reduce the migration cost later by forcing your to promisify/hookify parts of the app now